scala/private/rule_impls.bzl

```diff
@@ -394,11 +395,27 @@ def _compile_or_empty(
         jars2labels,
         implicit_junit_deps_needed_for_java_compilation,
         unused_dependency_checker_mode,
-        unused_dependency_checker_ignored_targets):
+        unused_dependency_checker_ignored_targets,
+        deps_providers = []):
```

Comment:
why aren't we passing it in to the scala_binary? targets can depend on the binary (I think it's also supported in java_binary)

Reply:
Yup you're right, I forgot to pass deps_providers from scala_binary. I hope that was what you meant.

Reply:
can we remove the default argument? Currently it might cause bugs (like the one we found)

Reply:
Yes, you're right. Removed the default value. Initially I needed it to test parts of the code before migrating the other usages.